plugin.py

```diff
@@ -1,158 +1,24 @@
-        if not java_path :
-            window.status_message("Please install java or set the 'java_home' setting")
-            return False
-
-        server_version = plugin_settings.get('server_version')
-        if not server_version :
-            window.status_message("'server_version' setting should be set")
-            return False
-
-        return True
-
-    def on_initialized(self, client) -> None:
-        register_client(client)
-
-
-def register_client(client):
-    client.on_notification(
-        "metals/status",
-        lambda params: on_metals_status(params))
-    client.on_notification(
-        "metals/executeClientCommand",
-        lambda params: on_execute_client_command(params))
-    client.on_request(
-        "metals/inputBox",
-        lambda params, request_id: on_metals_inputbox(client, params, request_id))
-
-def on_metals_inputbox(client, params, request_id):
-    def send_response(input: 'Optional[str]' = None):
-        param = { 'value': input, 'cancelled': False } if input else {'cancelled': True}
-        client.send_response(Response(request_id, param))
-
-    sublime.active_window().show_input_panel(
-        params.get('prompt', ''),
-        params.get('value', ''),
-        lambda value: send_response(value),
-        None,
-        lambda: send_response(None)
+if os.path.basename(os.path.dirname(__file__)) != "LSP-metals":
+    name = os.path.basename(os.path.dirname(__file__))
+    fmt = (
+        'The directory of the LSP-metals package is named "{}".',
+        'Please rename it to "LSP-metals" for correct functioning of',
+        'this package.'
     )
-
-
-def on_metals_status(params):
-    view = sublime.active_window().active_view()
-    hide = params.get('hide', False)
-    if(hide):
-        view.erase_status(server_name)
-    else:
-        view.set_status(server_name, params.get('text', ''))
-
-def _open_file(location: Location):
-    file_path = location.file_path
-    row = location.range.start.row + 1
-    col = location.range.start.col + 1
-    encoded_file_name = "{}:{}:{}".format(file_path, row, col)
-    window = sublime.active_window()
-    window.open_file(encoded_file_name, sublime.ENCODED_POSITION)
-
-def on_execute_client_command(params):
-    # handle https://scalameta.org/metals/docs/editors/new-editor.html#goto-location-1
-    if(params.get('command') == "metals-goto-location"):
-        args = params.get('arguments', [])
-        if(args):
-            location = Location.from_lsp(args[0])
-            _open_file(location)
-
-def prepare_server_properties(properties: 'List[str]') -> 'List[str]':
-    stripped = map(lambda p: p.strip(), properties)
-    none_empty = list(filter(None, stripped))
-    return none_empty
+    sublime.error_message(" ".join(fmt).format(name))
+else:
+    try:
+        from LSP.plugin import __version__ as v
+        if v < (1, 1, 0):
+            from .st3 import *
+        elif (1, 1, 0) <= v < (1, 2, 0):
+            from .st4 import *
```

Comment:
by rules of semantic versioning, the "st4" API should work until 2.0.0, not until 1.2.0.